presto-main/src/main/java/com/facebook/presto/execution/scheduler/NodeSchedulerConfig.java

```diff
@@ -34,6 +35,7 @@
     private boolean includeCoordinator = true;
     private int maxSplitsPerNode = 100;
     private int maxPendingSplitsPerTask = 10;
+    private int maxUnacknowledgedSplitsPerTask = 500;
```

Comment:
Just curious, what is the rationale behind using 500 as the default value? 🤔

Reply:
In an experimental setup I was running with differently (very large) split queue depth configurations and small files (aka: cheap splits that are processed very quickly once delivered), 500 seemed to be about the point of diminishing returns in terms of setting this value higher.

In the original iteration, I had this default to `Integer.MAX_VALUE` since I don't aim to affect any of the default scheduling behavior with this change, but rather just want to add a safety net for deeper split queues blowing up task update sizes. Dain convinced me to pick a more "reasonable" value that was still above the ceiling of having any affect on the current default configs.